miio/viomivacuum.py

```diff
@@ -422,22 +717,228 @@ def set_dnd(
+        return self.send("del_map", [map_id])
+
+    @command(
+        click.argument("map_id", type=int),
+        click.argument("map_name", type=str),
+    )
+    def rename_map(self, map_id: int, map_name: str):
+        """Rename map."""
+        maps = self.get_maps()
+        if map_id not in [m["id"] for m in maps]:
+            raise ViomiVacuumException("Map id {} doesn't exists".format(map_id))
+        return self.send("rename_map", {"mapID": map_id, "name": map_name})
+
+    @command(
+        click.option("--map-id", type=int, default=None),
+        click.option("--map-name", type=str, default=None),
+        click.option("--refresh", type=bool, default=False),
+    )
+    def get_rooms(
+        self, map_id: int = None, map_name: str = None, refresh: bool = False
+    ):
+        """Return room ids and names."""
+        if self._cache["rooms"] and not refresh:
+            return self._cache["rooms"]
+        if map_name:
+            map_id = None
+            maps = self.get_maps()
+            for map_ in maps:
+                if map_["name"] == map_name:
+                    map_id = map_["id"]
+            if map_id is None:
+                raise ViomiVacuumException(
+                    "Error: Bad map name, should be in {}".format(
+                        ", ".join([m["name"] for m in maps])
+                    )
+                )
+        elif map_id:
+            maps = self.get_maps()
+            if map_id not in [m["id"] for m in maps]:
+                raise ViomiVacuumException(
+                    "Error: Bad map id, should be in {}".format(
+                        ", ".join([str(m["id"]) for m in maps])
+                    )
+                )
+        # https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/blob/d73925c0106984a995d290e91a5ba4fcfe0b6444/index.js#L969
+        # https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum#semi-automatic
+        schedules = self.send("get_ordertime", [])
+        # ['1', '1', '32', '0', '0', '0', '1', '1', '11', '0', '1594139992', '2', '11', 'ami', '13', 'cuisine']
+        # [id, enabled, repeatdays, hour, minute, ?, ? , ?, ?, ?, ?, nb_of_rooms, room_id, room_name, room_id, room_name, ...]
+        # Find ALL specific scheduled cleanupS containing the room ids
+        rooms = {}
+        scheduled_found = False
+        for raw_schedule in schedules:
+            schedule = raw_schedule.split("_")
+            # Scheduled cleanup needs to be scheduled for 00:00 and inactive
+            if schedule[1] == "0" and schedule[3] == "0" and schedule[4] == "0":
+                scheduled_found = True
+                raw_rooms = schedule[12:]
+                rooms_iter = iter(raw_rooms)
+                rooms.update(
+                    dict(itertools.zip_longest(rooms_iter, rooms_iter, fillvalue=None))
+                )
```

Comment:
I suggest creating a new method (`parse_schedule`) returning a schedule item in wanted format.

Reply:
fixed